### ISSUE TITLE
Mesh.PRIMITIVE_TRIANGLE_FAN seems to have been removed in Godot 4

### ISSUE THREAD

**Issue by blackears** [godotengine/godot#72775]
### Godot version

v4.0.beta17.official [c40020513]

### System information

Windows 10, Compatability

### Issue description

Porting some old code that uses Mesh.PRIMITIVE_TRIANGLE_FAN.  Triangle fans seem to have been removed from Godot 4.  This is causing problems for me because I am drawing 2D convex shapes for which triangle fans are the most efficient way to pack vertices.

Triangle fans were one of the basic ways OpenGL handled mesh vertex data.  Given how useful they are in 2D processing, it would be nice to have them added back in as one of the vertex formats.

### Steps to reproduce

Try to use Mesh.PRIMITIVE_TRIANGLE_FAN in a GDScript in Godot 4.

### Minimal reproduction project

N/A

**Comment by AThousandShips** [comment]
To get triangles from triangle fan input you can use `SurfaceTool.add_triangle_fan` which generates triangles from a fan, for convenience, not sure why fans were removed but I suspect there's some implementation related reason

**Comment by clayjohn** [comment]
I think Triangle fans were removed because they are not well supported by modern graphics APIs like Vulkan 

This is the commit that removed the option https://github.com/clayjohn/godot/commit/9b0dd4f571ff431e23b9097e7f29746f4157be12#diff-e84616c8a1b4f2029ec6cc009fb3ed17f43d699e1f9125d8023c5ce9b65e66a8

Have you profiled your code to see if the triangle fan was actually more performant? If I recall correctly, many modern drivers internally unpack the Vertex fan into an array of triangles like what we do with SurfaceTool.add_triangle_fan.

**Comment by blackears** [comment]
No, I haven't profiled it and at this point it would probably be easier to just convert everything to triangle strips.

**Comment by AThousandShips** [comment]
Triangle fans are bad for cache, which seems to be why they've been deprecated in various graphics systems
